podcasts/Onboarding/Plus/PlusCoordinator.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import SwiftUI
+
+class PlusCoordinator: ObservableObject {
+    var navigationController: UINavigationController? = nil
+
+    func unlockTapped() {
+        guard let navigationController else { return }
+
+
+        let backgroundColor = UIColor(hex: PlusPurchaseModal.Config.backgroundColorHex)
+        let modal = PlusPurchaseModal(coordinator: PlusPurchaseCoordinator())
+        let controller = MDCSwiftUIWrapper(rootView: modal, backgroundColor: backgroundColor)
+        controller.presentModally(in: navigationController)
+    }
+
+    func dismissTapped() {
+        navigationController?.dismiss(animated: true)
+    }
+}
+
+extension PlusCoordinator {
+    static func make(in navigationController: UINavigationController? = nil) -> UIViewController {
```

Comment:
I think we can apply something similar to what I suggested in the previous PR. So we can try to decouple view, coordinator/model, and ViewController.